Pagination for area and climb media

glassbead0 opened the pull request:
Add query endpoints for `areaMediaPagination` and `climbMediaPagination`

For each query, the uuid is required (`climbUuid` or `areaUuid`), and the other pagination options `after`, `first`,  and `maxFiles`

Returns the uuid (climb or area), and a `mediaConnection`, with the same structure as the `getUserMediaPagination` endpoint, including totalItems.

Technical details:
For queuing areas/climbs with lots of media (eg, the entire US), this _should_ be efficient and fairly fast.
We only query the number of media objects + 1 requested with `first` (defaults to 6), so we aren't loading all the media objects into memory. We use `countDocuments` to efficiently get the totalItems.

Testing performance locally is difficult as i don't currently have an efficient way to add lots and lots of media.

will it be possible to deploy this to staging and performance test there before pushing to production?
@vnugent @Vichy97 


![Screenshot 2025-02-17 at 2 08 28 PM](https://github.com/user-attachments/assets/4b8d49cf-f76e-47ec-a0bc-afddc7ee0f64)
![Screenshot 2025-02-17 at 2 09 00 PM](https://github.com/user-attachments/assets/565055c1-df90-4fec-898d-557e053d58b2)

glassbead0 committed:
feat: pagination endpoints for areas and climbs

Vichy97 commented:
> Add query endpoints for `areaMediaPagination` and `climbMediaPagination`
> 
> For each query, the uuid is required (`climbUuid` or `areaUuid`), and the other pagination options `after`, `first`,  and `maxFiles`
> 
> Returns the uuid (climb or area), and a `mediaConnection`, with the same structure as the `getUserMediaPagination` endpoint, including totalItems.
> 
> Technical details:
> For queuing areas/climbs with lots of media (eg, the entire US), this _should_ be efficient and fairly fast.
> We only query the number of media objects + 1 requested with `first` (defaults to 6), so we aren't loading all the media objects into memory. We use `countDocuments` to efficiently get the totalItems.
> 
> Testing performance locally is difficult as i don't currently have an efficient way to add lots and lots of media.
> 
> will it be possible to deploy this to staging and performance test there before pushing to production?
> @vnugent @Vichy97 
> 
> 
> #TODO:
> Write unit tests.
> 
> I wanted to start the draft and get any feedback on the code for now.
> 
> ![Screenshot 2025-02-17 at 2 08 28 PM](https://github.com/user-attachments/assets/4b8d49cf-f76e-47ec-a0bc-afddc7ee0f64)
> ![Screenshot 2025-02-17 at 2 09 00 PM](https://github.com/user-attachments/assets/565055c1-df90-4fec-898d-557e053d58b2)
> 

The problem is that staging doesn't have images. I think deploying to prod is fine though since nobody will be using the new endpoints yet

glassbead0 commented:
I finished writing tests for the new endpoints. 

I haven't modified the existing `area` and `climb` endpoints yes to allow paginated media within the existing queries. I don't remember why we needed that. @vnugent Can you remind me? I'm happy to do it, but want to make sure its a useful implementation before i put work into it.

vnugent commented:
@glassbead0 we talked about creating a new GraphQL field, eg `mediaPagination`.   Once added I'd be happy to merge and deploy to staging.  [USA](https://stg.openbeta.io/area/1db1e8ba-a40e-587c-88a4-64f5ea814b8e/usa) on staging does have 181 photos (I just added a bunch of test photos and more if needed).

glassbead0 committed:
embedded area/climb media pagination

glassbead0 committed:
feat: improve MediaDataSource test

glassbead0 commented:
> @glassbead0 we talked about creating a new GraphQL field, eg `mediaPagination`. Once added I'd be happy to merge and deploy to staging. [USA](https://stg.openbeta.io/area/1db1e8ba-a40e-587c-88a4-64f5ea814b8e/usa) on staging does have 181 photos (I just added a bunch of test photos and more if needed).

I added new fields to the area and climb API's.  There is now a new field under `climb` and `area` called `mediaPagination`, which has the same interface as the top level `areaMediaPagination` and `climbMediaPagination`, except they  dont' require a UUID (since that is specified at the top level of the `climb`/`area` api and would be redundant)

I see an `areas` test under the integration tests for testing the area api, but currently it doesn't test anything around media, and there doesn't appear to be a `climb` integration test at all. I will take a look tomorrow to see how feasible it is to add tests for these new mediaPagination fields, and also welcome input on whether/how those tests should be implemented.
![Screenshot 2025-02-18 at 4 29 49 PM](https://github.com/user-attachments/assets/357884ab-ea7f-4d87-8714-67f26f7adbd3)
![Screenshot 2025-02-18 at 4 29 56 PM](https://github.com/user-attachments/assets/37d9975e-2281-423b-9051-24879e7f48b0)

glassbead0 committed:
feat: area mediaPagination test

glassbead0 commented:
I've added a test for the mediaPagination within the existing `area` query.  I won't have time to add a full new test for the `climb` query (since there isn't an existing one) until mid-late next week.

I'm happy to let this sit until then if consensus is that we should add a test before merging, but seems like adding a more comprehensive test of the climb endpoint (along with several other endpoints that dont' have integration tests), should maybe be separate tasks?

What are your thoughts @vnugent
